galore/cross_sections.py

```diff
@@ -281,3 +303,242 @@ def _eval_fit(energy, coeffs):
+
+def _cross_sections_from_csv_data(energy,data,dataset):
+    """
+    Get cross sections from data dict 
+    For known sources, data is based on tabulation of Yeh/Lindau (1985).[1]
+    Otherwise, energies in keV from 1-1500 are used with log-log polynomial
+    parametrisation of data from Scofield.[2]
+    
+    References:
+      1. Yeh, J.J. and Lindau, I. (1985)
+         Atomic Data and Nuclear Data Tables 32 pp 1-155
+      2. J. H. Scofield (1973) Lawrence Livermore National Laboratory
+         Report No. UCRL-51326
+    
+    Args:
+        energy(float): energy value  
+        data(dict): data from read_csv_file()
+        reference(str): 'Scofield' or 'Yeh'
+        
+    Returns:
+        orbitals_cross_sections_dict: containing orbitals 's', 'p', 'd', 'f' and 
+                                      maximum cross sections of each orbital. 
+                                      Missing data is represented as None.
+  
+    """   
+
+    n_subshells = len(data['electron_counts'])
+    subshells_headers = data['headers'][-n_subshells:]
+    cross_sections_by_subshells = data['data_table'].T[-n_subshells:]
+    
+    ##build dicts for easy calculation. 
+    electron_counts_by_subshells = dict(zip(subshells_headers,data['electron_counts']))
+    cross_sections_by_subshells = dict(zip(subshells_headers,data['data_table'].T[-n_subshells:]))
+
+    ## match the import energy
+    energy_index = np.abs(data['data_table'].T[0] - float(energy)).argmin()
+
+    ## build result dict
+    orbitals_cross_sections_dict = {}
+    
+    ## result for s orbital
+    s_cross_sections = np.array([value[energy_index] for key, value in cross_sections_by_subshells.items() if 's' in key])
+    electrons_number = np.array([value for key, value in electron_counts_by_subshells.items() if 's' in key])
+    ## get unit cross sections of s orbital
+    unit_cross_sections = s_cross_sections/electrons_number
+    ## get highest cross section of obital s
+    highest_obital_cross_section = np.max(np.nan_to_num(unit_cross_sections))
+    orbitals_cross_sections_dict['s'] = highest_obital_cross_section
+    
+    
+    
+    ## result for 'p', 'd', 'f' orbitals
+    orbitals = ['p', 'd', 'f']
+    
+    for orbital in orbitals:
+        interm_matrix =  np.array([value for key, value in cross_sections_by_subshells.items() if orbital in key]).T
+        electrons_number  = np.array([value for key, value in electron_counts_by_subshells.items() if orbital in key])
+ 
+        if np.shape(interm_matrix) != (0,):
+            if dataset == 'Scofield':
+                subshells_cross_sections = interm_matrix[energy_index]
+                result = _get_avg_orbital_cross_sections(subshells_cross_sections,electrons_number)
+         
+                ## get highest cross section of this obital
+                highest_obital_cross_section = np.max(np.nan_to_num(result))
+                orbitals_cross_sections_dict[orbital] = highest_obital_cross_section
```

Comment:
The correct orbital is not necessarily the one with the largest _value_, it's the one with the largest _index_.